### ISSUE TITLE
new a11y rule: accessible headings

### ISSUE THREAD

**Issue by HamletDRC** [Microsoft/tslint-microsoft-contrib#274]
All heading elements (h1..h6) should have text content
    • https://oaa-accessibility.org/rule/39/
    • https://github.com/globant-ui/arialinter/blob/master/lib/rules/headingsAndLabels.js

Text content for a heading must not come just from image alt text. (http://oaa-accessibility.org/wcag20/rule/19/)

Heading content should be concise. (http://oaa-accessibility.org/wcag20/rule/20/)

Heading elements (h1..h6) should be used for structuring information on the page. (http://oaa-accessibility.org/wcag20/rule/23/)

Missing or empty H1 element. (http://oaa-accessibility.org/wcag20/rule/14/)

H1 element content should not come only from the alt text of an image. (http://oaa-accessibility.org/wcag20/rule/15/)

H1 should match a subset of the words in the title element. (http://oaa-accessibility.org/wcag20/rule/16/)

No more than two h1 elements. (http://oaa-accessibility.org/wcag20/rule/17/)

**Comment by noamyogev84** [comment]
Hi @JoshuaKGoldberg, I'd like to try this one out.
possible?

**Comment by JoshuaKGoldberg** [comment]
Absolutely, please do!

**Comment by noamyogev84** [comment]
Hi @JoshuaKGoldberg, @HamletDRC 
In order to better understand this rule can you please help with the following questions:
1. What is a11y?
2. *"Heading content should be concise.."* how can I determine?
3. *"Heading elements (h1..h6) should be used for structuring information on the page..."*, how do you enforce this?

Thanks in advance!

**Comment by JoshuaKGoldberg** [comment]
> 1. What is a11y?

Accessibility. Ironic that the term for making things more accessible is a subtly inaccessible term. 🤷

> 2 _"Heading content should be concise.." how can I determine?

The link in the OP no longer has any real info and I can't find anything else on the internet... How about limiting to a maximum number of characters or to a single sentence? Maximum number of characters would be easiest to implement as a configurable rule; keeping to a single sentence is [surprisingly difficult](https://github.com/JoshuaKGoldberg/sentences-per-line/blob/master/index.js).

> _"Heading elements (h1..h6) should be used for structuring information on the page..."_, how do you enforce this?

Maybe it's that if a heading comes after another heading, the number must never increase by more than 1?

```tsx
// Ok
<h1>One</h1>
<p>...</p>
<h2>Two</h2>
<p>...</p>
```
```tsx
// Not OK
<h1>One</h1>
<p>...</p>
<h3>Three</h3>
<p>...</p>
```


If it's not that, I have no idea. 😄
